File: task_equivalents/llm_translations/gemini/optimized/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import pandas as pd
from sklearn.model_selection import train_test_split

College = pd.read_csv('data/csv/College.csv', index_col=0)

College['Private01'] = (College['Private'] == 'Yes').astype(int)
College = College.drop(columns=['Private'])

X = College.drop(columns=['Apps']).apply(pd.to_numeric)
Y = College['Apps']

xtrain, xtest, ytrain, ytest = train_test_split(X, Y, test_size=0.5, random_state=1)

File: task_equivalents/llm_translations/gemini/optimized/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
College <- read.csv('data/csv/College.csv', row.names = 1)

College$Private01 <- as.integer(College$Private == 'Yes')
College$Private <- NULL

X <- subset(College, select = -Apps)
X[] <- lapply(X, as.numeric)
Y <- College$Apps

set.seed(1)
train_idx <- sample(seq_len(nrow(X)), size = nrow(X) %/% 2)
xtrain <- X[train_idx, ]
xtest  <- X[-train_idx, ]
ytrain <- Y[train_idx]
ytest  <- Y[-train_idx]